state-chain/traits/src/mocks/egress_handler.rs

```diff
@@ -74,7 +74,7 @@ impl<C: Chain> EgressApi<C> for MockEgressHandler<C> {
 		destination_address: <C as Chain>::ChainAccount,
 		maybe_ccm_with_gas_budget: Option<(CcmDepositMetadata, <C as Chain>::ChainAmount)>,
 	) -> Result<ScheduledEgressDetails<C>, DispatchError> {
-		if amount.is_zero() {
+		if amount.is_zero() && maybe_ccm_with_gas_budget.is_none() {
```

Comment:
this is a good catch. we should add a test for this (could be another PR)

Reply:
This is just the mock - behaviour in the runtime was already correct.